src/Collection.php

```diff
@@ -921,7 +958,14 @@ public function insertOne($document, array $options = [])
             $options['writeConcern'] = $this->writeConcern;
         }
 
-        $operation = new InsertOne($this->databaseName, $this->collectionName, $document, $options);
+        $options = $this->handleCodecAndTypeMapOptions($options);
+
+        $operation = new InsertOne(
```

Comment:
That's an interesting point - I haven't even considered that. It would certainly reduce the boilerplate here and allow us to contain the logic in the operations themselves. I'll give this some more thought 👍